src/directives/message.js

```diff
@@ -1,55 +1,30 @@
 angular.module('schemaForm').directive('sfMessage',
-['$injector', 'sfErrorMessage', function($injector, sfErrorMessage) {
+['sfErrorMessage', function(sfErrorMessage) {
   return {
     scope: false,
     restrict: 'EA',
-    link: function(scope, element, attrs) {
-
-      //Inject sanitizer if it exists
-      var $sanitize = $injector.has('$sanitize') ?
-                      $injector.get('$sanitize') : function(html) { return html; };
-
-      //Prepare and sanitize message, i.e. description in most cases.
-      var msg = '';
-      if (attrs.sfMessage) {
-        msg = scope.$eval(attrs.sfMessage) || '';
-        msg = $sanitize(msg);
-      }
-
-      var update = function(valid) {
-        if (valid && !scope.hasError()) {
-          element.html(msg);
-        } else {
-
-          var errors = Object.keys(
-            (scope.ngModel && scope.ngModel.$error) || {}
-          );
-
-          // We only show one error.
-          // TODO: Make that optional
-          var error = errors[0];
-
-          if (error) {
-            element.html(sfErrorMessage.interpolate(
-              error,
-              scope.ngModel.$modelValue,
-              scope.ngModel.$viewValue,
-              scope.form,
-              scope.options && scope.options.validationMessage
-            ));
-          } else {
-            element.html(msg);
-          }
-        }
-      };
-      update();
-
-      scope.$watchCollection('ngModel.$error', function() {
-        if (scope.ngModel) {
-          update(scope.ngModel.$valid);
-        }
+    template: '<div ng-if="!error && form.description" ng-bind-html="form.description"></div><div ng-if="error" ng-bind-html="error"></div>',
+    controller: ['$scope', function($scope) {
```

Comment:
It would actually be better written as link – I'll go ahead and do that. My first iteration actually used both (one to compile the element, the other the way it is). I'm going to change this back to link :) good catch!